stbtrace/iterative_template.py

```diff
@@ -0,0 +1,167 @@
+#
+# Copyright 2019 Delphix. All rights reserved.
+#
+# SPDX-License-Identifier: GPL-2.0-or-later
+#
+# The IterativeTemplate class builds on python string templates to
+# allow iteration to substitute multiple lines for a single
+# template variable as well as simple substitutions.
+#
+# For example, consider the case where following string is the input
+# teamplateString used to initialize the objest. Then the render
+# method is called the first and last line would bass through
+# unchanged. The three central lines indicate an iterative substituion
+# will be performed.  A "hists" group should have been added will
+# 1 or field dict added to the group.  Each dict is expected to
+# contain the fields "name" and "aggtype".   A line will be generated
+# for each field substituting the values in the dictionary for the
+# specified fields.
+#
+# pretemplate string
+# $hists:{hist|
+# helper.add_am.group(2)("$hist.name$", BCCHelper.$hist.aggtype$)
+# }$
+# postemplate string
+#
+
+import re
+from string import Template
+
+
+class IterativeTemplate:
+    #
```

Comment:
please use python docstring format for class intro block comments (and also for public methods).

Reply:
I relocated the class level comment and used the docstring.  

The comments explaining the use of the regular expressions I left as is.  
That seemed to follow python convention but wasn't sure about comments on class level constants.  +